packages/client/lib/service/txpool.ts

```diff
@@ -178,27 +192,148 @@ export class TxPool {
     }
   }
 
+  /**
+   * Returns the gas tip (the max priority fee) for each transaction type
+   * @param tx - tx to check
+   * @returns - max priority fee
+   */
+  private txGasTipCap(tx: TypedTransaction): BN {
+    switch (tx.type) {
+      case 0:
+        return (tx as Transaction).gasPrice
+      case 1:
+        return (tx as AccessListEIP2930Transaction).gasPrice
+      case 2:
+        return (tx as FeeMarketEIP1559Transaction).maxPriorityFeePerGas
+      default:
+        throw new Error(`tx of type ${tx.type} unknown`)
+    }
+  }
+
+  /**
+   * This returns the gas fee cap (max fee per gas) for each transaction type
+   * @param tx - tx to check
+   * @returns - max fee per gas
+   */
+  private txGasFeeCap(tx: TypedTransaction): BN {
+    switch (tx.type) {
+      case 0:
+        return (tx as Transaction).gasPrice
+      case 1:
+        return (tx as AccessListEIP2930Transaction).gasPrice
+      case 2:
+        return (tx as FeeMarketEIP1559Transaction).maxFeePerGas
+      default:
+        throw new Error(`tx of type ${tx.type} unknown`)
+    }
+  }
+
+  private validateTxGasBump(existingTx: TypedTransaction, addedTx: TypedTransaction) {
+    const existingTipCap = this.txGasTipCap(existingTx)
+    const existingFeeCap = this.txGasFeeCap(existingTx)
+    const newTipCap = this.txGasTipCap(addedTx)
+    const newFeeCap = this.txGasTipCap(addedTx)
+    const minTipCap = existingTipCap.add(existingTipCap.muln(MIN_GAS_PRICE_BUMP_PERCENT).divn(100))
+    const minFeeCap = existingFeeCap.add(existingFeeCap.muln(MIN_GAS_PRICE_BUMP_PERCENT).divn(100))
+    if (newTipCap.lt(minTipCap) || newFeeCap.lt(minFeeCap)) {
+      throw new Error('replacement gas too low')
+    }
+  }
+
+  /**
+   * Validates a transaction against the pool and other constraints
+   * @param tx The tx to validate
+   */
+  private async validate(tx: TypedTransaction, isLocalTransaction: boolean = false) {
+    if (!tx.isSigned()) {
+      throw new Error('Attempting to add tx to txpool which is not signed')
+    }
+    if (tx.data.length > TX_MAX_DATA_SIZE) {
+      throw new Error(
+        `Tx is too large (${tx.data.length} bytes) and exceeds the max data size of ${TX_MAX_DATA_SIZE} bytes`
+      )
+    }
+    // This is the GasPrice which the miner receives: miner does not want
+    // to mine underpriced txs where miner gets almost no fees
+    const currentGasPrice = this.txGasTipCap(tx)
+    if (!isLocalTransaction) {
+      const txsInPool = this.txsInPool
+      if (txsInPool >= MAX_POOL_SIZE) {
+        throw new Error('Cannot add tx: pool is full')
+      }
+      // Local txs are not checked against MIN_GASPRICE
+      if (currentGasPrice.lt(MIN_GASPRICE)) {
+        throw new Error(`Tx does not pay the minimum gas price of ${MIN_GASPRICE.toString()}`)
```

Comment:
```suggestion
        throw new Error(`Tx does not pay the minimum gas price of ${MIN_GASPRICE}`)
```
you don't need to call `toString()` in template literals it will automatically call it for you. it would be great if you can also remove in the other places in this file.